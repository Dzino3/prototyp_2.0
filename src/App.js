import React, { useState } from 'react';
import LoginForm from './components/LoginForm';
import OrderForm from './components/OrderForm';
import NavbarForm from './components/NavbarForm';
import AvailabilityForm from './components/AvailabilityForm';
import ListForm from './components/ListForm';
import HeroForm from './components/HeroForm';
import FooterForm from './components/FooterForm';
import './components/style/appStyle.css';
import Axios from "axios";


function App() {
  const adminUser = { name: "admin123" }
  const [user, setUser] = useState({ acc: '' , unit: '', dealer: ''});
  const [error, setError] = useState("");


  const Login = details => {
    if(details.acc === adminUser.name){
      setUser({
        acc: details.acc
      });
    }
    Axios.post("https://codux.herokuapp.com/login", { ACC: details.acc, UNIT: details.unit }).then((response) => {
      if (response.data.length === 0 || details.acc !== response.data[0].ACC) {
        setError("Wrong Input")
      } else {

        setUser({
          acc: details.acc,
          unit:  details.unit,
          dealer: response.data[0].DEALER
        });
      }
    });
  }

  const Logout = details => {
    setUser({ acc: "" });
  }

  return (
    <>
      
      <div className="App">
        {(user.acc !== "") ? (
          <div className='website'>
            <div>
              <NavbarForm Logout={Logout}/>
            </div>
            <div className='pageContent'>
              
              <div className='mainContent'>
                <HeroForm user={user}></HeroForm>
                <AvailabilityForm user={user}></AvailabilityForm>
                <OrderForm user={user} error={error} setError={setError}></OrderForm>
                <ListForm user={user}></ListForm>
              </div>
            </div>
            <FooterForm></FooterForm>
          </div>
        ) : (
          <LoginForm Login={Login} error={error}></LoginForm>
        )
        }
      </div>
    </>
  );
}


export default App;
