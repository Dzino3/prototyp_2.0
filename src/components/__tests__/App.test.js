import App from '../../App';
import React from "react";
import renderer from 'react-test-renderer';

describe('something truthy and falsy', () => {
    test.skip('true to be true', () => {
      expect(true).toBe(true);
    });
  
    test.skip('false to be false', () => {
      expect(false).toBe(false);
    });
  });

  describe(`App Rendering`, () => {

    test('should render App', function () {
        let tree = renderer.create(<App/>);

        expect(tree.toJSON()).toMatchSnapshot();
    });
})