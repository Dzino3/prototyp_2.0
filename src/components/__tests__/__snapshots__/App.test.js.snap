// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Rendering should render App 1`] = `
<div
  className="App"
>
  <div
    className="loginFormWhole"
  >
    <div
      className="backImg"
    />
    <div
      className="fuckYou"
    >
      <div
        className="logoImg"
      />
      <form
        className="loginForm"
        onSubmit={[Function]}
      >
        <div
          className="form-inner"
        >
          <h2
            className="loginH2"
          >
            Welcome to the Tracker
          </h2>
          <select
            className="countrySelect"
            id="country"
            name="country"
            onChange={[Function]}
          >
            <option>
              Select your (unit) country
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Austria
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Belgium
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Bulgaria
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Croatia
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Cyprus
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Czech Republic
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Denmark
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Estonia
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Finland
            </option>
            <option
              className="unitOption"
              value="22"
            >
              (22) France
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) French Guiana
            </option>
            <option
              className="unitOption"
              value="27"
            >
              (27) Germany
            </option>
            <option
              className="unitOption"
              value="31"
            >
              (31) Great Britain
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Greece
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Hungary
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Iceland
            </option>
            <option
              className="unitOption"
              value="31"
            >
              (31) Ireland
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Israel
            </option>
            <option
              className="unitOption"
              value="41"
            >
              (41) Italy
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Kazakhstan
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Latvia
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Lithuania
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Luxembourg
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Netherlands
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Norway
            </option>
            <option
              className="unitOption"
              value="4P"
            >
              (4P) Poland
            </option>
            <option
              className="unitOption"
              value="32"
            >
              (32) Portugal
            </option>
            <option
              className="unitOption"
              value="32"
            >
              (32) Romania
            </option>
            <option
              className="unitOption"
              value="3R"
            >
              (3R) Russia/Kalinigrad
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Serbia
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Slovakia
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Slovenia
            </option>
            <option
              className="unitOption"
              value="32"
            >
              (32) Spain
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Sweden
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Switzerland
            </option>
            <option
              className="unitOption"
              value="48"
            >
              (48) Ukraine
            </option>
          </select>
          <div
            className="loginForm-group"
          >
            <label
              className="loginLabel"
            >
              Account number:
            </label>
            <input
              className="loginInput"
              id="acc"
              name="acc"
              onChange={[Function]}
              type="text"
              value=""
            />
            <input
              className="loginButton"
              type="submit"
              value="Login"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
